akka-http/client/src/main/scala/endpoints/akkahttp/client/AkkaHttpClient.scala

```diff
@@ -0,0 +1,126 @@
+package endpoints.akkahttp.client
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.stream.Materializer
+import akka.stream.scaladsl.{Flow, Sink, Source}
+import endpoints.algebra.{Decoder, Encoder, MuxRequest}
+import endpoints.{Tupler, algebra}
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+final case class AkkaHttpClientSettings(
+  requestExecutor: AkkaHttpRequestExecutor,
+  baseUri: Uri = Uri("/"),
+  toStrictTimeout: FiniteDuration = 2.seconds,
+  stringContentExtractor: HttpEntity.Strict => String = _.data.utf8String
+)
+
+trait AkkaHttpRequestExecutor {
+  def apply(request: HttpRequest): Future[HttpResponse]
+}
+
+object AkkaHttpRequestExecutor {
+  def cachedHostConnectionPool(host: String, port: Int)(implicit system: ActorSystem, materializer: Materializer): AkkaHttpRequestExecutor =
+    default(Http().cachedHostConnectionPool[Int](host, port))
+
+  def default(poolClientFlow: Flow[(HttpRequest, Int), (Try[HttpResponse], Int), Http.HostConnectionPool])(implicit materializer: Materializer): AkkaHttpRequestExecutor =
+    new AkkaHttpRequestExecutor {
+      override def apply(request: HttpRequest): Future[HttpResponse] =
+        Source.single(request -> 1)
+          .via(poolClientFlow)
+          .map(_._1.get)
+          .runWith(Sink.head)
+    }
+}
+
+
+class AkkaHttpClient(val settings: AkkaHttpClientSettings)
+                    (implicit val EC: ExecutionContext, val M: Materializer) extends algebra.Endpoints with Urls with Methods {
```

Comment:
So far the naming convention is to use the same name for an algebra interface and its implementation (but the interface lives in the `algebra` package, whereas the implementation lives in a different package. Would that convention work for you too?

Reply:
Makes sense, will change!